### PR TITLE
[API] add GET /orgs endpoint

### LINKED ISSUE #5361: missing list endpoint for /orgs/
- Gitea version (or commit ref): 1.5.3
- Git version:
- Operating system:
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description

the api endpoints for orgs and teams are missing list options. I would like to get a list of all team ids. Either by querying "GET /teams/" and get all teams (or all team ids) or by querying "GET /orgs/" to get all orgs and then query each org for it's teams.

### PR DESCRIPTION
**Pull request by 6543**
close #5361

add `GET /orgs` endpoint: wich return list of all public organizations

### CODE REVIEW

File: templates/swagger/v1_json.tmpl [hunk 1]
```diff
@@ -468,6 +468,35 @@
...
     },
     "/orgs": {
+      "get": {
+        "produces": [
+          "application/json"
+        ],
+        "tags": [
+          "organization"
+        ],
+        "summary": "Get all public organizations",
```
Comment (bagasme) [inline templates/swagger/v1_json.tmpl]: Only public orgs?

Comment (6543) [reply templates/swagger/v1_json.tmpl]: yes for now ...

**Comment by 6543** [comment]
@bagasme 
> Only public orgs?

now if user is loged in he will find Limited Orgs also and if he is admin he will get realy all

**Review by zeripath — APPROVED** [review]

File: models/user.go [hunk 2]
```diff
@@ -1492,8 +1492,9 @@ func (opts *SearchUserOptions) toConds() builder.Cond {
 		cond = cond.And(keywordCond)
 	}
 
-	if !opts.Private {
-		// user not logged in and so they won't be allowed to see non-public orgs
+	if len(opts.Visible) > 0 {
+		cond = cond.And(builder.In("visibility", opts.Visible))
+	} else {
```
Comment (lunny) [inline models/user.go:1497]: I think to remove the `else` is better?

Comment (6543) [reply models/user.go:1497]: what do you mean? else ... default value (show only public)

Comment (lunny) [reply models/user.go:1497]: As a consideration for a compitable with before, it's also OK.

File: integrations/api_org_test.go [hunk 3]
```diff
@@ -136,3 +136,17 @@ func TestAPIOrgDeny(t *testing.T) {
...
+
+	req := NewRequestf(t, "GET", "/api/v1/orgs")
+	resp := MakeRequest(t, req, http.StatusOK)
+
+	var apiOrgList []*api.Organization
+	DecodeJSON(t, resp, &apiOrgList)
+
+	assert.Len(t, apiOrgList, 7)
+	assert.Equal(t, "org25", apiOrgList[0].FullName)
+	assert.Equal(t, "public", apiOrgList[0].Visibility)
```
Comment (lunny) [inline integrations/api_org_test.go:151]: How about also add a failed test?

Comment (6543) [reply integrations/api_org_test.go:151]: this should not fail ... it return public orgs and if you are loged in it return limited orgs too ...

Comment (6543) [reply integrations/api_org_test.go:151]: @lunny :arrow_up:

**Review by lunny — APPROVED** [review]

**Pull request merged**